### ISSUE TITLE
Collection data loop in one specific service

### ISSUE THREAD

**Issue by javixeneize** [a85/POSTMan-Chrome-Extension#3339]
Hi

This is not an issue, its more a question about a feature that i dont know if it is available

I can run a loop from a data file (adding a data file in collection runner and mapping it to a variable, as explained in https://www.getpostman.com/docs/postman/collection_runs/working_with_data_files)

My question is... I have a project that contains 3 services. Lets call it A,B and C. 

I want to run that data loop 4 times only in service C, so  i want the execution to be A,B,C,C,C,C. Can it be done?

If i set the iteration option to 4, what i am running is 4 times the whole project, it is, A,B,C,A,B,C,A,B,C,A,B,C

Thanks

**Comment by sdnts** [comment]
Hi @javixeneize [postman.setNextRequest](https://www.getpostman.com/docs/postman/collection_runs/building_workflows) should be able to do what you want.

Essentially what you'd want is a global variable (counter) that keeps track of how many times the request has run. AT the end of the request (in the test script), you'd have to increment the counter and `setNextRequest` to the same request if the counter is less than 4.

Please reopen if this does not solve your problem!

**Comment by javixeneize** [comment]
Hi

Its near to what i want, but not exactly that. I can run this "C" service now in a loop using the setNextRequest, but, the second part of this is not working for me

I have a data file with a fixed number of rows, I want to run this "C" service in a loop with the info in the data file for that subsequent iteration

It is, if i have a datafile with 4 rows, and data is 1,2,3,4... I want it to run "C" those 4 times, mapping this value there

So, in short, i can run a loop on a specific service, and on the other side, i can do a loop reading from a data file mapping this value to a field in the request, but, i cant run both things together, which is what i need 

Thanks
